lib/core/presentation/const/header.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class Header{
+
+  //Top Bar
+  AppBar headerBar(BuildContext context){
+    return AppBar(
+      // iconTheme: IconThemeData(
+      //   color: Colors.black, //change your color here
+      // ),
+      elevation: 0,
+      backgroundColor: Colors.white,
+      title: const DeallLogo(0.125,0.02),
+      centerTitle: true,
+    );
+  }
+
+  //Font type
+  TextStyle deallFont(BuildContext context, Color c){
+    return GoogleFonts.cantarell(
+      textStyle: Theme.of(context).textTheme.headline5,
+      // fontSize: d,
+      color: c,
+      fontWeight: FontWeight.w100,
+    );
+  }
+
+}
+
+//LOGO
+class DeallLogo extends StatelessWidget{
+  // final double d;
+  final double w;
+  final double h;
+  
+
+  const DeallLogo(this.w, this.h);
+  // const DeallLogo(this.d, this.w, this.h);
+
+  @override
+  Widget build(BuildContext context){
+    final mq = MediaQuery.of(context);
+
+    return Container(
+      width: mq.size.width * w,
+      height: mq.size.height * h,
+      child: FittedBox(
+        fit: BoxFit.fitWidth,
+        child: RichText(
+          text: TextSpan(
+            text: 'DEAL',
+            style: Header().deallFont(context, Colors.black),
+            children:  <TextSpan>[
+              TextSpan(
+                text: 'L', 
+                style: Header().deallFont(context, Colors.red)
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+} 
```

Comment:
The way u structure wrong already. Each widget should be in a separate file and widget file shouldn't be store in a const folder.

For example, Dealllogo can be stored in a file called deall_logo,dart.
AppBar can be another widget too.

You dont have to try hard to extract the widgets, the purpose of extraction is mainly so that we can reuse the widget in other place or the widget is complex enough that u think it can be a separate widget. So in this case, if your appbar will only be used in welcome page, then dont extract.

Const folder is to store const values that will not be change throught the app.

Reply:
okay i'll separate the deal logo to another file. the appbar will be used for other pages also eh. 

would storing in core/widgets be ideal?

Reply:
the app logo can be stored in core/presentation/widget

as for the Header class i dont think it is needed, just make ur custom AppBar widget into a stateless or stateful widget